src/test/java/com/bakdata/kafka/ErrorLoggingFlatTransformerTopologyTest.java

```diff
@@ -0,0 +1,311 @@
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(
+                        (Transformer<? super Object, ? super Object, ? extends Iterable<Object>>) null))
+                .isInstanceOf(NullPointerException.class);
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(
+                        (Transformer<? super Object, ? super Object, ? extends Iterable<Object>>) null, filterAll()))
+                .isInstanceOf(NullPointerException.class);
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(
+                        (TransformerSupplier<? super Object, ? super Object, ? extends Iterable<Object>>) null))
+                .isInstanceOf(NullPointerException.class);
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(
+                        (TransformerSupplier<? super Object, ? super Object, ? extends Iterable<Object>>) null,
+                        filterAll()))
+                .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void shouldNotAllowNullFilter(final SoftAssertions softly) {
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(this.mapper, null))
+                .isInstanceOf(NullPointerException.class);
+        softly.assertThatThrownBy(() -> ErrorLoggingFlatTransformer.logErrors(() -> this.mapper, null))
+                .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void shouldForwardSchemaRegistryTimeout(final SoftAssertions softly) {
+        final RuntimeException throwable = createSchemaRegistryTimeoutException();
+        this.mapper = new Transformer<>() {
+            private ProcessorContext context = null;
+
+            @Override
+            public void init(final ProcessorContext context) {
+                this.context = context;
+            }
+
+            @Override
+            public Iterable<KeyValue<Double, Long>> transform(final Integer key, final String value) {
+                if (key.equals(1) && "foo".equals(value)) {
+                    throw throwable;
+                }
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void close() {
+
+            }
+        };
+        this.createTopology();
+        softly.assertThatThrownBy(() -> this.topology.input()
+                        .withValueSerde(STRING_SERDE)
+                        .add(1, "foo"))
+                .hasCauseInstanceOf(SerializationException.class);
+        final List<ProducerRecord<Double, Long>> records = Seq.seq(this.topology.streamOutput(OUTPUT_TOPIC)
+                        .withKeySerde(DOUBLE_SERDE)
+                        .withValueSerde(LONG_SERDE))
+                .toList();
+        softly.assertThat(records)
+                .isEmpty();
+    }
+
+    @Test
+    void shouldNotCaptureThrowable(final SoftAssertions softly) {
```

Comment:
The test name is a little confusing, because this tests `Error` and not `Throwable`. We capture `Throwable` if it's a `RuntimeException`